aten/src/ATen/core/ivalue_inl.h

```diff
@@ -522,6 +522,25 @@ c10::optional<T> generic_to(
   return std::move(ivalue).to<T>();
 }
 
+template <typename Tuple, std::size_t... I>
+Tuple generic_to_tuple_impl(
+    const std::vector<IValue>& t,
+    c10::guts::index_sequence<I...>) {
+  return std::make_tuple(
+      t[I].to<typename std::tuple_element<I, Tuple>::type>()...);
+}
+
+template <
+    typename... Args,
+    typename Indices = c10::guts::make_index_sequence<sizeof...(Args)>,
+    c10::guts::enable_if_t<
+        !c10::guts::disjunction<std::is_lvalue_reference<Args>...>::value,
+        std::nullptr_t> = nullptr>
```

Comment:
It's probably better if we use SFINAE based on `is_constructible<IValue, Args>...`, if it works.